### DISCUSSION TITLE
Wishlist: SQL-Query generator

### DISCUSSION THREAD

**Discussion by cleder** [dineug/erd-editor#260] [Ideas]
Is generating Queries (like MS Access) on the roadmap or a possible future feature? For me that would be the Killer application feature ;-)

**Comment by dineug** [comment]
What an interesting feature!
It wasn't on the roadmap.
I'll add it.

Thanks for letting me know.

**Comment by cleder** [comment]
A possible use case I was thinking about was to use it in superset https://github.com/apache/incubator-superset instead of or in addition to e.g: https://superset.apache.org/sqllab.html

**Reply by cleder** [reply]
[graphic-walker](https://github.com/Kanaries/graphic-walker) is another candidate, one could also try to integrate it with this to show the ERD

**Comment by cleder** [comment]
Do you know of https://github.com/mWater/jsonql? This seems to be a nice intermediate format/API specification (alternatives may be available, that is just the only one I could find)

**Comment by dineug** [comment]
It's easy for anyone to use it
It's a standard web interface.
It was the use case I thought.

An example of that was vscode extension.

But I think it's not enough if you don't fully support reverse engineering with SQL DDL.

The https://github.com/mWater/jsonql you told me about
I'll refer to it when I develop it. Thank you!

**Comment by cleder** [comment]
Sorry, this may be OT, but I thought, I use this issue to link interesting stuff related to this here.

https://github.com/danielrearden/sqlmancer

Sqlmancer is a Node.js library that empowers you to effortlessly and efficiently translate GraphQL queries into SQL statements. 
Multiple dialect support. Sqlmancer supports Postgres, MySQL, MariaDB and SQLite, enabling you to incorporate it into existing projects regardless of what flavor of SQL you’re using.
Performance. Avoid the N+1 problem by building a single SQL…

I do not want to distract you from your current great work, quite on the contrary, keep it up :-)

**Comment by cleder** [comment]
> But I think it's not enough if you don't fully support reverse engineering with SQL DDL.

Is there an intermediate JSON format?

Does the full DDL to VUERD conversion pipeline have to be implemented here?

I can think of existing introspection tools that may be adopted to spit out some well-defined JSON.

**Comment by dineug** [comment]
Intermediate json format does not exist

The conversion pipeline may be implemented externally.

What I'm thinking right now is...

[sql-ddl-to-json-schema](https://github.com/duartealexf/sql-ddl-to-json-schema) contributing
or
[syntax-parser](https://github.com/ascoders/syntax-parser) use

Or to take advantage of the existing grammar.
WebAssembly is thinking, too.

I haven't thought about this yet, so I think it'll be clear when it's developed.

**Comment by cleder** [comment]
The 'SQL DDL to JSON Schema converter' looks promising, I like the compact format. Another pro for a format like this would be that the target could also be an ORM like SQLalchemy, Hibernate or typeorm.io - many others are also available

**Comment by cleder** [comment]
https://github.com/xnuinside/simple-ddl-parser also defines a DDL dialect agnostic json schema

**Comment by cleder** [comment]
http://dataprotocols.org/json-table-schema/
JSON Table Schema
This RFC defines a simple schema for tabular data. The schema is designed to be expressible in JSON.

**Comment by dineug** [comment]
#452
